File: Bmi_2/app/src/main/java/com/pjatk/bmi_2/model/QuizResult.java
package com.pjatk.bmi_2.model;

import android.arch.persistence.room.ColumnInfo;
import android.arch.persistence.room.Entity;
import android.arch.persistence.room.PrimaryKey;

import com.pjatk.bmi_2.R;

import java.util.Date;

@Entity(tableName = "quizresult")
public class QuizResult {
    @PrimaryKey(autoGenerate = true)
    public int id;
    @ColumnInfo(name = "correct")
    public int correct;
    @ColumnInfo(name = "wrong")
    public int wrong;
    @ColumnInfo(name = "percent")
    public int percent;
    @ColumnInfo(name = "date")
    public Date date;

    public QuizResult(){

    }

    public QuizResult(int correct, int wrong)
    {
        this.date = new Date();
        CalculatePercent(correct,wrong);
    }

    private void CalculatePercent(int correct,int wrong){
        this.correct = correct;
        this.wrong = wrong;
        int sum = correct + wrong;
        double p = (double)correct/(double) sum;
        p *= 100;
        percent = (int)p;
    }

    public String getColor(){
        if(percent >= 51)
            return "#4CAF50";
        else
            return "#E91E36";
    }
}

File: BMICalculator/app/src/main/java/com/szymon/bmicalculator/model/QuizResult.kt
package com.szymon.bmicalculator.model

import android.arch.persistence.room.ColumnInfo
import android.arch.persistence.room.Entity
import android.arch.persistence.room.PrimaryKey
import java.util.*

@Entity(tableName = "quizresult")
class QuizResult(correct: Int, wrong: Int) {
    @PrimaryKey(autoGenerate = true)
    var id: Int = 0
    @ColumnInfo(name = "correct")
    var correct: Int = 0
    @ColumnInfo(name = "wrong")
    var wrong: Int = 0
    @ColumnInfo(name = "percent")
    var percent: Int = 0
    @ColumnInfo(name = "date")
    var date: Date? = null

    val color: String
        get() = if (percent >= 51)
            "#4CAF50"
        else
            "#E91E36"

    init {
        this.date = Date()
        calculatePercent(correct, wrong)
    }

    private fun calculatePercent(correct: Int, wrong: Int) {
        this.correct = correct
        this.wrong = wrong
        val sum = correct + wrong
        var p = correct.toDouble() / sum.toDouble()
        p *= 100.0
        percent = p.toInt()
    }
}